src/x11api.c

```diff
@@ -88,49 +88,73 @@ int cxevent(Display *display, long mask, XButtonEvent event)
     return TRUE;
 }
 
-int click(Display *display, int button, int mode)
+XButtonEvent make_event(Display *display, int button)
+{
+    XButtonEvent event;
+    memset(&event, 0, sizeof(event));
+    event.button = button;
+    event.same_screen = True;
+    event.subwindow = DefaultRootWindow(display);
+
+    while (event.subwindow)
+    {
+        event.window = event.subwindow;
+        XQueryPointer(display, event.window,
+                        &event.root, &event.subwindow,
+                        &event.x_root, &event.y_root,
+                        &event.x, &event.y,
+                        &event.state);
+    }
+    return event;
+}
+
+int mouse_down(Display *display, int button, int mode)
```

Comment:
Why have 2 almost identical functions. Instead make one called something like mouse_event, then it accepts an enum value of UP or DOWN.

```c
enum MouseEvents
{
    MOUSE_UP,
    MOUSE_DOWN,
};


int mouse_event(..., enum MouseEvents event_type) {
}
```

Reply:
you're right. I should have think about that. If you don't fix it, I'll tomorrow. Or this night if I can't sleep for whatever reason.